frontend/app/assets/javascripts/spree/frontend/views/spree/shared/variant_select.js

```diff
@@ -0,0 +1,87 @@
+let getQueryString = window.location.search
+let urlParams = new URLSearchParams(getQueryString)
+let variantIdFromUrl = urlParams.get('variant')
+
+this.initializeQueryParamsCheck = function () {
+  if (urlParams.has('variant')) {
+    this.variants.forEach(verifyVariantID)
```

Comment:
I guess there should be only one `for` loop. You have now O(n^2) time complexity.

Reply:
Fixed this issue.